pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -477,50 +447,104 @@ func TestReconcile(t *testing.T) {
 	}
 
 	for _, tc := range testCases {
-		t.Run(tc.testCase, func(t *testing.T) {
-			var objects []runtime.Object
-			objects = append(objects, tc.mhc)
-			if tc.machine != nil {
-				objects = append(objects, tc.machine)
-			}
-			objects = append(objects, tc.node)
-			recorder := record.NewFakeRecorder(2)
-			r := newFakeReconcilerWithCustomRecorder(recorder, objects...)
+		recorder := record.NewFakeRecorder(2)
+		r := newFakeReconcilerWithCustomRecorder(recorder, buildRunTimeObjects(tc)...)
+		assertBaseReconcile(t, tc, ctx, r)
```

Comment:
This (and the other loops in this file) still need to be wrapped in a `t.Run` otherwise there is only one test, not a parent with children. If we do it like this, the first test failure will fail the whole table, and it will be hard to work out exactly which one.

Please restore the `t.Run` here and in the external remediation test too

Reply:
Hi @JoelSpeed,
Thanks Great comment :+1: 
As far as I could see the other loop was already ok so I I've only changed the one you've commented about.
```for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			recorder := record.NewFakeRecorder(2)
			r := newFakeReconcilerWithCustomRecorder(recorder, buildRunTimeObjects(tc)...)
			assertBaseReconcile(t, tc, ctx, r)
			//assertExternalRemediation
			assertExternalRemediation(t, tc, ctx, r)

		})
	}```